pkg/config/remote/client/client.go

```diff
@@ -46,6 +46,12 @@ type ConfigFetcher interface {
 	ClientGetConfigs(context.Context, *pbgo.ClientGetConfigsRequest) (*pbgo.ClientGetConfigsResponse, error)
 }
 
+// Listener defines the interface of a remote config listener
+type Listener interface {
+	OnUpdate(map[string]state.RawConfig, func(cfgPath string, status state.ApplyStatus))
+	OnStateChange(bool)
```

Comment:
Although thinking more on this - this also now leaves OnUpdate ambiguous as well since we now cover multiple forms of updates. So given that it also needs an adjustment I think it's fine to cover that separately in another PR.